### ISSUE TITLE
Impossible to use version 2.0.0 of @org.eclipse.jdt.annotation.NonNull and @org.eclipse.jdt.annotation.Nullable

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#362]
```
What steps will reproduce the problem?
1.Create in Eclipse a project with compliance JavaSE-1.8.
2.Create the following java file
abstract class Foo<E> {
    E e;

    void bar() {
        e = null;
    }

    abstract <F> Foo<@NonNull F> foo(F newValue);
}
3. The compiler complains that @NonNull cannot be resolved to a type. With 
Ctrl+1, choose the Quick Fix "Copy library with default null annotation to 
build path".
(import org.eclipse.jdt.annotation.NonNull; should be also automatically 
inserted)
4. Ctrl+Click on @NonNull. You should see that the target of @NonNull is 
@Target({ TYPE_USE }), which means that you are really using the version 2.0.0 
of the JDT annotations (and not the version 1.1)
5. Run the Nullness checker adding the version *2.0.0* of the JDT annotations 
to the classpath

What is the expected output? What do you see instead?
Expected: A warning is issued for line 
e = null;
Actual: A compiler error for line
abstract <F> Foo<@NonNull F> foo(F newValue)

What version of the product are you using? On what operating system?
checker-framework: 1.8.5
Eclipse: 4.4.0 (Luna)
Java: 1.8
org.eclipse.jdt.annotations: 2.0.0

Please provide any additional information below.

```

Original issue reported on code.google.com by `ClovisSe...@gmail.com` on 25 Sep 2014 at 7:54
